pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AvroSchema.java

```diff
@@ -44,27 +44,32 @@
-        reflectDataAllowNull.addLogicalTypeConversion(new TimeConversions.TimeConversion());
+            reflectDataAllowNull.addLogicalTypeConversion(new Conversions.DecimalConversion());
+            reflectDataAllowNull.addLogicalTypeConversion(new TimeConversions.DateConversion());
+            reflectDataAllowNull.addLogicalTypeConversion(new TimeConversions.LossyTimeMicrosConversion());
+            reflectDataAllowNull.addLogicalTypeConversion(new TimeConversions.LossyTimestampMicrosConversion());
+            reflectDataAllowNull.addLogicalTypeConversion(new TimeConversions.TimeMicrosConversion());
+            reflectDataAllowNull.addLogicalTypeConversion(new TimeConversions.TimestampMicrosConversion());
+            reflectDataAllowNull.addLogicalTypeConversion(new TimeConversions.TimestampConversion());
+            reflectDataAllowNull.addLogicalTypeConversion(new TimeConversions.TimeConversion());
 
-        ReflectData reflectDataNotAllowNull = ReflectData.get();
+            ReflectData reflectDataNotAllowNull = ReflectData.get();
 
-        reflectDataNotAllowNull.addLogicalTypeConversion(new Conversions.DecimalConversion());
-        reflectDataNotAllowNull.addLogicalTypeConversion(new TimeConversions.DateConversion());
-        reflectDataNotAllowNull.addLogicalTypeConversion(new TimeConversions.TimestampConversion());
-        reflectDataNotAllowNull.addLogicalTypeConversion(new TimeConversions.LossyTimeMicrosConversion());
-        reflectDataNotAllowNull.addLogicalTypeConversion(new TimeConversions.LossyTimestampMicrosConversion());
-        reflectDataNotAllowNull.addLogicalTypeConversion(new TimeConversions.TimeMicrosConversion());
-        reflectDataNotAllowNull.addLogicalTypeConversion(new TimeConversions.TimestampMicrosConversion());
-        reflectDataNotAllowNull.addLogicalTypeConversion(new TimeConversions.TimeConversion());
+            reflectDataNotAllowNull.addLogicalTypeConversion(new Conversions.DecimalConversion());
+            reflectDataNotAllowNull.addLogicalTypeConversion(new TimeConversions.DateConversion());
+            reflectDataNotAllowNull.addLogicalTypeConversion(new TimeConversions.TimestampConversion());
+            reflectDataNotAllowNull.addLogicalTypeConversion(new TimeConversions.LossyTimeMicrosConversion());
+            reflectDataNotAllowNull.addLogicalTypeConversion(new TimeConversions.LossyTimestampMicrosConversion());
+            reflectDataNotAllowNull.addLogicalTypeConversion(new TimeConversions.TimeMicrosConversion());
+            reflectDataNotAllowNull.addLogicalTypeConversion(new TimeConversions.TimestampMicrosConversion());
+            reflectDataNotAllowNull.addLogicalTypeConversion(new TimeConversions.TimeConversion());
+        } catch (Throwable t) {
+            LOG.warn("Avro logical types are not available. If you are going to use avro logical types, " +
+                    "you can include `joda-time` in your dependency.");
```

Comment:
@codelipenghui One last ask. I missed this one. Can we have this message as debug instead of warn? Otherwise it would print the warning for everyone, even if not interested in using Joda-Time types with Avro.

Reply:
Makes sense.